### DISCUSSION TITLE
Is it possible to have multiple instances of useForm() and share their context with any child?

### DISCUSSION THREAD

**Discussion by arthurmmedeiros** [react-hook-form/react-hook-form#9324] [Q&A]
Hello,

I have a sort of wizard form where I can submit each step separatelly. Some steps however need data from other steps, so I'm trying to find a way to be able to use `useFormContext` to access the data I need. `FormProvider` seemed to be the solution, but in my case I have multiple instances of useForm(), and I can't assign these instances to a single `FormProvider`.

Initially I was trying to expose the values to the parent, but it is kind of hard to keep the values synced. 

```
interface IForm1 {
    id?: number;
}

interface IForm2 {
    name?: string;
}

const Component1 = () => {
    const formContext1 = useFormContext<IForm1>();
    const formContext2 = useFormContext<IForm2>();

    const id = formContext1.watch('id');
    const name = formContext2.watch('name');

    return <div>
        {id}
        and
        {name}
    </div>;
};
```
I thought of doing something like this, but the deepest FormProvider would always override the parent FormProviders.
```
const MyForms = () => <>
    <FormProvider {...form1}>
        <FormProvider {...form2}>
            <GenericForm<IForm1>>
                <Component1 />
            </GenericForm>
            <GenericForm<IForm2>>
                <Component1 />
            </GenericForm>
        </FormProvider>
</>;
```

Any ideias?

**Comment by arthurmmedeiros** [comment]
I found a way to combine the instances of useForm()

It is something like this:

```
const form1 = useForm();
const form2 = useForm();


<FormProvider {...{ ...form1, ...form2}}>
   <form>Form 1</form>
   <form>Form 2</form>
</FormProvider>
```

The problem is that when I try to use `useFormContext` I can see that the context contains a large object with all the properties of the two forms as if like they were simply combined.

I thought the contexts would be separated from each other and that I could do something like:
`const form1 = useFormContext<IForm1>();`

This does not work =/

**Reply by npearson72** [reply]
Well, `{...{ ...form1, ...form2}}` just merges the both forms into a single object, thereby overriding form1 with form2.

Try: `console.log({...{...{a: 1}, ...{a: 2}}})`

Maybe create your own `FormsProvider` and pass multiple forms as an array (or object with form names)?

**Reply by arthurmmedeiros** [reply]
I found a way to do this and published it [here](https://medium.com/@tu.mmedeiros/creating-a-multi-independent-steps-form-with-react-hook-form-447206750600)
